### PR TITLE
Making RandomErasing Out of place by default and improvising tests for the same

### PR DESCRIPTION
**Pull request by surgan12**
This PR includes:
   1. Making Random Erasing out of place by default.
   2. Test for checking the non-alteration of the unerased portion.
   3. Test for verifying the optional in-place behaviour 

 
cc: @fmassa

### CODE REVIEW

File: test/test_transforms.py [hunk 1]
```diff
@@ -1345,11 +1345,19 @@ def test_random_grayscale(self):
...
         # Test Set 1: Erasing with int value
-        img_re = transforms.RandomErasing(value=0)(img)
-        assert img_re.size(0) == 3
+        img_re = transforms.RandomErasing(value=0)
+        i, j, h, w, v = img_re.get_params(img, scale=img_re.scale, ratio=img_re.ratio, value=img_re.value)
+
+        # Check if the unerased region is preserved
+        img_output = F.erase(img, i, j, h, w, v)
+        erased_region = torch.zeros([3, 60, 60], dtype=torch.float32)
+        erased_region[:, i:i + h, j:j + w] = v
```
Comment (fmassa) [inline test/test_transforms.py]: Looks like `erased_region` is all zeros, as `v` is zero?

File: test/test_transforms.py [hunk 2]
```diff
@@ -1345,11 +1345,19 @@ def test_random_grayscale(self):
...
+
+        assert torch.equal(img - img_output, erased_region)
```
Comment (fmassa) [inline test/test_transforms.py]: if `erased_region` is all zeros, this means that `img` is equal to `img_output`.

Comment (surgan12) [reply test/test_transforms.py]: Yep, will change the value in the tests.

**Comment by surgan12** [comment]
@fmassa sure ! will make it out of place.

**Comment by surgan12** [comment]
@fmassa  ready for a review !!

**Pull request merged**